## 06 Confidantial Interval - KARAN JOLIYA

Scenario A manufacturer of print-heads for personal computers is interested in estimating the mean durability of their print-heads in terms of the number of characters printed before failure. To assess this, the manufacturer conducts a study on a small sample of print-heads due to the destructive nature of the testing process.

Data, A total of 15 print-heads were randomly selected and tested until failure. The durability of each print-head (in millions of characters) was recorded as follows:

1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29

## Assignment Tasks

##### a. Build 99% Confidence Interval Using Sample Standard Deviation
- Assuming the sample is representative of the population, construct a 99% confidence interval for the mean number of characters printed before the print-head fails using the sample standard deviation. Explain the steps you take and the rationale behind using the t-distribution for this task

In [2]:
import numpy as np
import math
from scipy import stats #Import libraries 

In [4]:
# Durability data (in millions of characters)
data = [ 1.13, 1.55, 1.43, 0.92, 1.25,1.36, 1.32, 0.85, 1.07, 1.48,1.20, 1.33, 1.18, 1.22, 1.29]
n = len(data)
n

15

In [6]:
sample_mean = np.mean(data) # Sample mean

sample_sd = np.std(data, ddof=1) # Sample standard deviation

print(f"Sample Mean : {sample_mean}")
print(f"Sample Standard Deviation :{sample_sd}") 

Sample Mean : 1.2386666666666666
Sample Standard Deviation :0.19316412956959936


In [7]:
confidence_level = 0.99
alpha = 1 - confidence_level
df = n - 1  # degree of freedom(df)

In [9]:
t_critical = stats.t.ppf(1 - alpha/2, df) # t critical value
print(f"t critical : {t_critical}")

t critical : 2.976842734370834


In [11]:
margin_error_t = t_critical * (sample_sd / math.sqrt(n)) # Margin of error
print(f"margin error : {margin_error_t}")

margin error : 0.14846932822817596


In [12]:
ci_t = (sample_mean - margin_error_t, sample_mean + margin_error_t ) # Confidence interval
print(f'confidence interval : {ci_t}')

confidence interval : (np.float64(1.0901973384384906), np.float64(1.3871359948948425))


Sample size is small (n=15) Population standard deviation is unknown Hence, t-distribution is used We are 99% confident that the true mean durability lies between 1.090 and 1.387 million characters

b. Build 99% Confidence Interval Using Known Population Standard Deviation
If it were known that the population standard deviation is 0.2 million characters, construct a 99% confidence interval for the mean number zof characters printed before failure.

In [14]:
population_sd = 0.2

z_critical = stats.norm.ppf(1 - alpha/2) # z critical value

margin_error_z = z_critical * (population_sd / math.sqrt(n)) # Margin of error

ci_z = (sample_mean - margin_error_z,sample_mean + margin_error_z) # Confidence interval

print(f"z critical : {z_critical}")
print(f"confidance interval : {ci_z}")

z critical : 2.5758293035489004
confidance interval : (np.float64(1.1056514133957607), np.float64(1.3716819199375725))


Population standard deviation is known  
Therefore, z-distribution is appropriate 
The interval is narrower because uncertainty is lower